### PR TITLE
Log Swap event through a low-level call

### LINKED ISSUE #43: Remove `EventBytes` struct and and emit `bytes` instead
### Problem

Emit bytes [here](https://github.com/pnetwork-association/pnetwork-v4/blob/master/solidity/src/contracts/Adapter.sol#L214-L225) insteand of EventBytes

### Notes

Issues may arise with `memory` typed arrays, the `bytes.concat` can't work. In case, consider emitting the event by manually concatenating the data into a fixed size array

### PR DESCRIPTION
**Pull request by envin3**
Fix #43 :Removed EventBytes struct from event emitter

### CODE REVIEW

File: solidity/test/hardhat/forked-environment/ptoken-upgrade.test.js [hunk 1]
```diff
@@ -142,12 +143,18 @@ conditionalDescribe(
...
           txId: swapEvent.transactionHash,
           originChainId: Chains.Bsc,
-          eventContent,
+          nonce: swapEvent.topics[1],
+          erc20: decodedEvent.erc20,
+          destinationChainId: decodedEvent.destination,
+          amount: decodedEvent.netAmount,
+          sender: decodedEvent.sender,
+          recipient: decodedEvent.recipient,
+          data: decodedEvent.data,
```
Comment (gitmp01) [inline solidity/test/hardhat/forked-environment/ptoken-upgrade.test.js:157]: cleaner for the reader eyes:
```suggestion
          ...decodedEvent,
          nonce: swapEvent.topics[1],
```

**Review by gitmp01 — CHANGES_REQUESTED** [review]

File: solidity/test/forge/Integration.t.sol [hunk 2]
```diff
@@ -301,6 +313,16 @@ contract IntegrationTest is Test, Helper {
...
+        for (uint256 i = 0; i < logs.length; i++) {
+            Vm.Log memory log = logs[i];
+
+            // Print log details
+            console.log("Log %d:", i);
+            // console.log("Address: %s", log.address);
+            console.logBytes(log.data);
+            console.logBytes32(log.topics[0]); // You can print multiple topics if necessary
+            console.logBytes32(log.topics[1]); // You can print multiple topics if necessary
+        }
```
[the remark this answers is not in the document]
Comment (envin3) [reply solidity/test/forge/Integration.t.sol:325]: debug code forgot there

File: solidity/test/forge/Helper.sol [hunk 3]
```diff
@@ -303,4 +301,37 @@ abstract contract Helper is Test, DeployHelper {
...
+            let dataStart := add(eventBytes, 32)
+            let length := mload(eventBytes)
+            log2(
+                dataStart,
+                length,
+                topic0,
+                topic1
+            )
+        }
+    }
```
Comment (gitmp01) [inline solidity/test/forge/Helper.sol:336]: :+1:

**Review by gitmp01 — APPROVED** [review]
LGTM

**Pull request merged**